src/nemos/utils.py

```diff
@@ -39,106 +38,181 @@
 
 
-def convolve_1d_trials(
-    basis_matrix: ArrayLike,
-    time_series: Union[Iterable[NDArray], NDArray, Iterable[jnp.ndarray], jnp.ndarray],
-) -> List[jnp.ndarray]:
-    """Convolve trial time series with a basis matrix.
+def check_convolve_input_dims(basis_matrix: jnp.ndarray, time_series: Any):
+    """
+    Check the dimensions of inputs for convolution operation.
 
-    This function checks if all trials have the same duration. If they do, it uses a fast method
-    to convolve all trials with the basis matrix at once. If they do not, it falls back to convolving
-    each trial individually.
+    This function validates that the `basis_matrix` is 2-dimensional and the `time_series`
+    is either a pytree of 2-dimensional arrays or a single 3-dimensional array.
 
     Parameters
     ----------
     basis_matrix :
-        The basis matrix with which to convolve the trials. Shape (window_size, n_basis_funcs).
+        A 2-dimensional array representing the basis matrix.
     time_series :
-        The time series of trials to convolve with the basis matrix. It should be a list of 2D arrays,
-        where each array represents a trial and its second dimension matches the first dimension
-        of the basis_matrix. Each trial has shape (n_timebins_trial, n_neurons).
-
-    Returns
-    -------
-    :
-        The convolved trials. It is a list of 3D arrays, where each array represents a convolved trial.
-        Each element of the list will have shape (n_timebins_trial - window_size - 1, n_neurons, n_basis_funcs).
+        A pytree of 2-dimensional arrays or a single 3-dimensional array representing the time series.
 
     Raises
     ------
     ValueError
-        - If basis_matrix is not a 2D array-like object.
-        - If trials_time_series is not an iterable of 2D array-like objects.
-        - If trials_time_series contains empty trials.
-        - If the number of time points in each trial is less than the window size.
+        If `basis_matrix` is not a 2-dimensional array or if `time_series` is not a pytree of
+        2-dimensional arrays or a single 3-dimensional array.
     """
-    basis_matrix = jnp.asarray(basis_matrix)
     # check input size
-    if basis_matrix.ndim != 2:
+    if not check_dimensionality(basis_matrix, expected_dim=2):
         raise ValueError("basis_matrix must be a 2 dimensional array-like object.")
-    elif 0 in basis_matrix.shape:
-        raise ValueError(
-            "Empty basis_matrix provided. "
-            f"The shape of basis_matrix is {basis_matrix.shape}!"
-        )
+
     try:
-        # this should fail for variable trial length
-        time_series = jnp.asarray(time_series)
         if time_series.ndim != 3:
-            raise ValueError
-
-    except ValueError:
-        # convert each trial to array
-        time_series = [jnp.asarray(trial) for trial in time_series]
+            raise AttributeError
```

Comment:
This should be `ValueError` to match docstring ,right? And have the same error message as below.